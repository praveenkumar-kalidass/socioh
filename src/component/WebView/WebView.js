import React from 'react';
import { Modal } from 'react-native';
import PropTypes from 'prop-types';
import RNWebView from 'react-native-webview';

import { ICON } from '../../constant';
import {
  ModalContainer,
  ModalHeader,
  HeaderIcon,
  SearchInput,
} from './WebView.style';

const WebView = ({ uri, onClose }) => {
  return (
    <Modal>
      <ModalContainer>
        <ModalHeader>
          <SearchInput testID="webview_uri" disabled value={uri} />
          <HeaderIcon
            testID="webview_close"
            type={ICON.MATERIAL_FAMILY}
            name={ICON.CLOSE}
            onPress={onClose}
          />
        </ModalHeader>
        <RNWebView source={{ uri }} />
      </ModalContainer>
    </Modal>
  );
};

WebView.propTypes = {
  uri: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default WebView;
