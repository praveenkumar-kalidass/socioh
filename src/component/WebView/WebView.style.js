import styled from 'styled-components/native';
import { Icon, Input } from 'native-base';

import { COLOR, SPACING } from '../../constant';

export const ModalContainer = styled.View`
  flex: 1;
`;

export const ModalHeader = styled.View`
  background-color: ${COLOR.PRIMARY};
  flex-direction: row;
  align-items: center;
  padding-horizontal: ${SPACING.SPACE_10};
`;

export const HeaderIcon = styled(Icon)`
  color: ${COLOR.WHITE};
`;

export const SearchInput = styled(Input)`
  color: ${COLOR.WHITE};
`;
