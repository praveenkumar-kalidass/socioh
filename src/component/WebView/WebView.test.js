import React from 'react';
import { render, act, fireEvent } from '@testing-library/react-native';

import WebView from './index';

describe('WebView', () => {
  const mockClose = jest.fn();
  const mockUri = 'https://github.com/praveenkumar-kalidass';

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const container = render(<WebView uri={mockUri} onClose={mockClose} />);

    expect(container).toMatchSnapshot();
  });

  it('should render uri', () => {
    const { getByTestId } = render(
      <WebView uri={mockUri} onClose={mockClose} />,
    );

    expect(getByTestId('webview_uri').props.value).toStrictEqual(mockUri);
  });

  it('should call close callback, when close icon is clicked', () => {
    const { getByTestId } = render(
      <WebView uri={mockUri} onClose={mockClose} />,
    );

    act(() => {
      fireEvent.press(getByTestId('webview_close'));
    });

    expect(mockClose).toHaveBeenCalledTimes(1);
  });
});
