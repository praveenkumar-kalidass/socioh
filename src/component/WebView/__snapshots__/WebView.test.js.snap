// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebView should match snapshot 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#1abfbf",
            "flexDirection": "row",
            "paddingHorizontal": 10,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        disabled={true}
        editable={false}
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
        testID="webview_uri"
        underlineColorAndroid="rgba(0,0,0,0)"
        value="https://github.com/praveenkumar-kalidass"
      />
      <Text
        allowFontScaling={false}
        onPress={[MockFunction]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#ffffff",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="webview_close"
        type="MaterialIcons"
      >
        Óóç
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <RNCWebView
        cacheEnabled={true}
        injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
        injectedJavaScriptForMainFrameOnly={true}
        javaScriptEnabled={true}
        messagingEnabled={false}
        onContentProcessDidTerminate={[Function]}
        onHttpError={[Function]}
        onLoadingError={[Function]}
        onLoadingFinish={[Function]}
        onLoadingProgress={[Function]}
        onLoadingStart={[Function]}
        onMessage={[Function]}
        onShouldStartLoadWithRequest={[Function]}
        source={
          Object {
            "uri": "https://github.com/praveenkumar-kalidass",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
            undefined,
          ]
        }
        useSharedProcessPool={true}
      />
    </View>
  </View>
</Modal>
`;
